DSCResources/MSFT_xDnsServerADZone/MSFT_xDnsServerADZone.psm1

```diff
@@ -68,7 +68,7 @@ function Get-TargetResource
         DynamicUpdate = $dnsServerZone.DynamicUpdate
         ReplicationScope = $dnsServerZone.ReplicationScope
         DirectoryPartitionName = $dnsServerZone.DirectoryPartitionName
-        Ensure = if ($dnsServerZone -eq $null) { 'Absent' } else { 'Present' }
+        Ensure = if ([bool]($dnsServerZone -eq $null)) { 'Absent' } else { 'Present' }
```

Comment:
The cast to a boolean isn't needed here.
Also, this won't pass PSScriptAnalyzer. The `$null` should be on the left side of the comparison instead of the right.
You can read more about it here: https://github.com/PowerShell/PSScriptAnalyzer/blob/development/RuleDocumentation/PossibleIncorrectComparisonWithNull.md